app/src/main/java/nerd/tuxmobil/fahrplan/congress/schedule/LectureViewDrawer.kt

```diff
@@ -90,11 +90,9 @@ internal class LectureViewDrawer(
         val eventIsFavored = event.highlight
         @ColorRes val backgroundColorResId: Int
         backgroundColorResId = if (eventIsFavored) {
-            val colorResId = trackNameBackgroundColorHighlightPairs[event.track]
-            colorResId ?: R.color.event_border_highlight
+            trackNameBackgroundColorHighlightPairs.getOrDefault(event.track, R.color.event_border_highlight)
```

Comment:
`getOrDefault()` is a platform method introduced with API 24 and Java 8 desugaring hasn't been enabled for the project (I verified that it crashes on an API 16 emulator).

You could use Kotlin's `getOrElse()`.

```kotlin
backgroundColorResId = if (eventIsFavored) {
    trackNameBackgroundColorHighlightPairs.getOrElse(event.track) { R.color.event_border_highlight }
} else {
    trackNameBackgroundColorDefaultPairs.getOrElse(event.track) { R.color.event_border_default }
}
```

Personally, I'd go for this:

```kotlin
@ColorRes val backgroundColorResId: Int = if (eventIsFavored) {
    trackNameBackgroundColorHighlightPairs[event.track] ?: R.color.event_border_highlight
} else {
    trackNameBackgroundColorDefaultPairs[event.track] ?: R.color.event_border_default
}
```

Reply:
Nice catch. I did not know that. Not very obvious. Fixed in ba3d0bafd182e8dd3ee392fe538f17592f870082.

Reply:
Simplified as suggested in a2c11726f49074eeeeefec05c638160d2c3b5d9b.

Reply:
The method name was displayed in red in my Android Studio (4.2 Canary 3).

Reply:
Same here. I thought this would be an IDE issue - as there are similar bugs from time to time. Sometimes they disappear by cleaning the cache - sometimes I am too lazy to do so. Seems this was not such a case.